src/ValidatorVisitor.php

```diff
@@ -348,19 +349,21 @@ public function leaveNode(Node $node){
                     /**
                      * @var Node\Stmt\UseUse    $use
                      */
-                    if($use instanceof Node\Stmt\UseUse && $use->name instanceof Node\Name && (is_string($use->alias) || is_null($use->alias))){
+                    if($use instanceof Node\Stmt\UseUse && $use->name instanceof Node\Name && (is_string($use->alias) || $use->alias instanceof Node\Identifier || is_null($use->alias))){
                         $this->sandbox->checkAlias($use->name->toString());
                         if($use->alias){
                             if(!$this->sandbox->checkKeyword('as')){
                                 $this->sandbox->validationError('Keyword failed custom validation!', Error::VALID_KEYWORD_ERROR, $node, 'as');
                             }
                         }
-                        $this->sandbox->defineAlias($use->name->toString(), $use->alias);
+                        $this->sandbox->whitelistClass($use->getAlias());
+                        $this->sandbox->whitelistType($use->getAlias());
```

Comment:
Note that I'm not sure if this is the correct change here, but I'm not sure how you can really enable aliases for unknown code without doing something like this.